packages/react-aria-components/src/Virtualizer.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {CollectionRenderer, CollectionRendererContext} from './Collection';
+import {DropTargetDelegate, Node} from '@react-types/shared';
+import {Layout, ReusableView, useVirtualizerState} from '@react-stately/virtualizer';
+import React, {ReactNode, useContext, useMemo} from 'react';
+import {useScrollView, VirtualizerContext, VirtualizerItem} from '@react-aria/virtualizer';
+
+export interface LayoutOptionsDelegate<O> {
+  useLayoutOptions?(): O
+}
+
+interface ILayout<O> extends Layout<Node<unknown>, O>, Partial<DropTargetDelegate>, LayoutOptionsDelegate<O> {}
+
+export interface VirtualizerProps {
+  children: ReactNode,
+  layout: ILayout<any>
+}
+
+export function Virtualizer(props: VirtualizerProps) {
+  let {children, layout} = props;
+  let renderer: CollectionRenderer = useMemo(() => ({
+    isVirtualized: true,
+    layoutDelegate: layout,
+    dropTargetDelegate: layout.getDropTargetFromPoint ? layout as DropTargetDelegate : undefined,
+    CollectionRoot({collection, focusedKey, scrollRef}) {
+      let layoutOptions = layout.useLayoutOptions?.();
+      let state = useVirtualizerState({
```

Comment:
I'm still digging to see where best to fix this, but I noticed that a virtualized table with renderEmptyState won't actually display the rendered empty state element because the wrapping div has a height of 0. It seems like that height of 0 is because that visible view (aka the body) doesn't have any items within it and thus it gets a layoutInfo calculated for it with a height of 0px.

Passing `enableEmptyState: true` to the TableLayout alleviates this somewhat (actually allocates a height), but the row element positioning still isn't quite right

Reply:
hmm yeah I didn't want the table to be responsible for rendering the empty state. I guess I can force the table body to always take up the entire height of the table so the empty state isn't clipped.

Reply:
that makes sense, that way the user can still control the styling. I was initially thinking on making `enableEmptyState` be true and have `CollectionBranch` take children but that was all kinds of janky